src/main.js

```diff
@@ -62,6 +62,16 @@ module.exports = {
     this.subscriptions = new CompositeDisposable()
     this.worker = null
 
+    // FIXME: Remove this after a few editions
```

Comment:
__*Preference*__

It might not always be clear what "a few editions" means.  It might be better to say something like "Remove this in 2019" (hopefully that would be enough time. ;)

Also, a small comment here about what this block of code is doing might be good, just to give context on why it exists.  In fact, we change rules once in a while, and I wonder if it might make sense to create a separate file/function that translates old rules to new ones.  It might help with organization, but then again maybe it's not worth it.  Up to you.

Reply:
> It might not always be clear what "a few editions" means.

I was thinking something along the lines of a few minor versions, but have nothing set in stone.

> I wonder if it might make sense to create a separate file/function that translates old rules to new ones.

I'd like to not be changing old options so often something like this is necessary, but the thought had crossed my mind.

Reply:
Yeah, in this case it's less about the number of versions, but the likelihood that users have opened Atom and run this code, which is why I thought a date constraint might make more sense.  NBD either way though.

Reply:
How about this? 
```
/** FIXME: Deprecated eslintRulesDir{String} option in version 8.3.3 in favor of 
  * eslintRulesDirs{Array<String>}. Remove handling of eslintRulesDir after a few
  * minor updates or before 2019.
  */
```

Reply:
I like that version @skylize, changed it a little bit in the upcoming change.